source/module_hamilt_general/operator.h

```diff
@@ -64,12 +64,14 @@ class Operator
 
     /// developer-friendly interfaces for act() function
     /// interface type 2: input and change the Psi-type HPsi
-    virtual void act(const psi::Psi<T, Device>& psi_in, psi::Psi<T, Device>& psi_out) const {};
+    // virtual void act(const psi::Psi<T, Device>& psi_in, psi::Psi<T, Device>& psi_out) const {};
+    virtual void act(const psi::Psi<T, Device>& psi_in, psi::Psi<T, Device>& psi_out, const int nbands) const {};
     /// interface type 3: return a Psi-type HPsi
     // virtual psi::Psi<T> act(const psi::Psi<T,Device>& psi_in) const { return psi_in; };
 
     Operator* next_op = nullptr;
 
+    virtual int get_act_type() const { return this->act_type; }
```

Comment:
1. Can you add annotations for functions in .h files?
2. How many act types of all Operators now and in planning? 
3. The disign of this virtual function should be described, how does developer to override it.

Reply:
Just 2 types now. And the function `get_act_type()` does not need to be virtual. 
See the change and annotations in the new commit.